app/core/views/submission.py

```diff
@@ -131,28 +126,51 @@ def edit_submission_view(request, pk=None, clone=False):
                 submission.challenge = container.challenge
                 submission.user = request.user
                 submission.save()
-                return redirect("submission-detail", pk=submission.pk)
+                arg_formset = ArgFormSet(
+                    request.POST, request.FILES, instance=submission
+                )
+                if arg_formset.is_valid():
+                    arg_instances = arg_formset.save(commit=False)
+                    for instance in arg_instances:
+                        instance.submission = submission
+                        instance.save()
+                    for instance in arg_formset.deleted_objects:
+                        instance.delete()
+                    return redirect("submission-detail", pk=submission.pk)
+                else:
+                    show_args = True
```

Comment:
@megosato this is the line that's supposed to handle the issue you mentioned on the previous PR:

> I have no idea where this would be handled:
but when I'm playing with the kvargs branch, if there's an error with special arguments, when I hit save: it minimizes the special arguments tab and opens the container and submission details tab,
... so its kind of hard to tell the special arguments is what is causing the issue

I'll add a couple more comments showing how this is used